python/cudf/cudf/core/index.py

```diff
@@ -1515,9 +1515,12 @@ def argsort(
         )
 
     def repeat(self, repeats, axis=None):
-        return self._from_columns_like_self(
+        res = self._from_columns_like_self(
             Frame._repeat([*self._columns], repeats, axis), self._column_names
         )
+        if isinstance(res, DatetimeIndex):
```

Comment:
Could you just override `DatetimeIndex.repeat` as

```python
def repeat(self, repeats, axis=None):
    res = super().repeats(repeats, axis=axis)
    res._freq = None
    return res
```

Reply:
Good idea, done 👍